### PR TITLE
enhance(tests): add cypress for e2e/integration tests

### PR DESCRIPTION
**Pull request by 5h1rU**
# Pull Request Checklist

You can go to [dendron pull requests](https://wiki.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html) to see full details for items in this checklist.

### Quality Assurance
- [x] add a [test](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#writing-tests) for the new feature
- [x] make sure all the existing [tests](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#running-all-tests) pass
- [x] do a spot check by running your feature with our [test workspace](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#test-workspace)
- [x] after you submit your pull request, check the output of our [integration test](https://github.com/dendronhq/dendron/actions) and make sure all tests pass
  - NOTE: if you running MacOS or Linux, pay special attention to the Windows output and vice versa if you are developing on Windows

#### Special Cases
- [ ] if your tests changes an existing snapshot, make sure that snapshots are [updated](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#updating-test-snapshots)
- [ ] if you are adding a new language feature (graphically visible in VS Code/preview/publishing), make sure that it is included in [test-workspace](https://wiki.dendron.so/notes/cb22bd36-d45a-4ffd-a31e-96c4b39cb7fb.html#test-workspace). We use this to manually inspect new changes and for auto regression testing

### Docs
- [x] Make sure that the PR title follows our [commit style](https://wiki.dendron.so/notes/adc39825-77a6-46cf-9c49-2642fcb4248e.html#commit-style)
- [x] Please summarize the feature or impact in 1-2 lines in the PR description
- [ ] If your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR

Example PR Description
```markdown
# feat: capitalize all foos

This changes capitalizes all occurrences of `foo` to `Foo`

Docs PR: <URL_TO_DOCS_PR>
```

### First Time PR
- [ ] sign the [CLA](https://en.wikipedia.org/wiki/Contributor_License_Agreement) which will be prompted by our github bot after you submit the PR
- [ ] add your [discord](https://discord.gg/AE3NRw9) alias in the review so that we can give you the [horticulturalist](https://wiki.dendron.so/notes/7c00d606-7b75-4d28-b563-d75f33f8e0d7.html#horticulturalist) badge in our community

### Analytics
- [ ] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### CODE REVIEW

File: .github/workflows/cypress-test.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Cypress Tests
```
Comment (kevinslin) [inline .github/workflows/cypress-test.yml:1]: instead of running as a separate test, can you add this to [ci.yml](https://github.com/dendronhq/dendron/blob/fix/move-header-new-note/.github/workflows/ci.yml#L115:L115)? 

since we don't need it to run on every os, you can use an if clause to gate it

```yml
- name: Run cypress 
  if: runner.os == 'Linux'
  uses: cypress-io/github-action@v2
  with:
      build: yarn run cypress-build
      start: yarn cypress-start 
```

File: .eslintrc.js [hunk 2]
```diff
@@ -83,7 +84,7 @@ module.exports = {
     // --- React
     "react/prop-types": "off",
     // we use 'logger' inside of hooks, gets flagged 
-    "react-hooks/exhaustive-deps": "off",
+    "react-hooks/exhaustive-deps": 1,
```
Comment (kevinslin) [inline .eslintrc.js:87]: does this generate compiler warnings and issues? if so, might be better to split off in a separate PR that also fixes the issues

File: packages/nextjs-template/cypress/integration/home.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+describe("Index", () => {
+  it("successfully loads", () => {
+    cy.visit("/");
```
Comment (kevinslin) [inline packages/nextjs-template/cypress/integration/home.ts:3]: i forget - does `cy.visit` crash if it can't load the page?

File: .github/workflows/cypress-test.yml (hunk 1 above)
Comment (kevinslin) [reply .github/workflows/cypress-test.yml:1]: after you do the new push, i can run the ci workflow manually from your branch which allows us to test this before merging to master

**Review by kevinslin — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 2 above)
Comment (5h1rU) [reply .eslintrc.js:87]: Good catch! This file is not part of this PR.

File: packages/nextjs-template/cypress/integration/home.ts (hunk 3 above)
Comment (5h1rU) [reply packages/nextjs-template/cypress/integration/home.ts:3]: It will throw a `CypressError` 

```cy.visit() failed trying to load:

http://localhost:3000/other

The response we received from your web server was:

  > 404: Not Found

This was considered a failure because the status code was not 2xx.

This http request was redirected '1' time to:

  - 308: http://localhost:3000/other/

If you do not want status codes to cause failures pass the option: failOnStatusCode: false```

**Review by kevinslin — APPROVED** [review]
